src/main/kotlin/Main.kt

```
@@ -0,0 +1,35 @@
+import calculator.CalculatorService
+import transactions.DateTime
+import transactions.TransactionService
+import utils.CsvReader
+import java.math.BigDecimal
+import java.text.NumberFormat
+
+private const val TRANSACTIONS_CSV = "transactions.csv"
+
+fun main() {
+    val accountId = input("accountId: ").uppercase()
+    val dateTimeFrom = DateTime(input("from: "))
+    val dateTimeTo = DateTime(input("to: "))
+
+    val ts = TransactionService(CsvReader.rows(TRANSACTIONS_CSV))
```

consider what would the user see if a file was not found. And I would consider adding a test for that scenario too. 
Also think about how we might handle that exception scenario, we might throw another exception and have it be handled somewhere else, or we might let it fail gracefully, depends on our UX in this app.

Ensure you are familiar with the Exceptions best practices in kotlin. They are different to in Java. Obviously Java has checked exceptions and unchecked. Kotlin does not have checked exceptions.

https://elizarov.medium.com/kotlin-and-exceptions-8062f589d07 Roman Elizarov is on the Kotlin team and used to be the tech lead of the language team itself. His post here is very insightful on where we should be considering exception handling in our Kotlin code.
Think about code you have seen where there are try/catch blocks polluting an entire chain of functions. And consider what he is saying here to keep that in one place, and let the exception bubble up gracefully.

---

just having a general understanding is ok for now, when you have Bitbucket access I'll show you some examples in the code that we have set up